### PR TITLE
Refactor stub content into independent, pluggable system

### PR DESCRIPTION
**Pull request by linc**
* Override stub content now possible.
* Multiple stub content now possible.
* Use static calls whenever possible.
* Do not `return` so we don't block subsequent structure items (since this is an `include`).

### CODE REVIEW

**Comment by tburry** [comment]
Is this necessary? We just want to change the stub content.

**Comment by linc** [comment]
You want to: 1) Have multiple discussions/comments with variable content like different authors and 2) Use cloud-specific content. So yeah, necessary.

File: applications/vanilla/settings/stub.php [hunk 1]
```diff
@@ -6,19 +6,12 @@
...
-$SQL = Gdn::Database()->SQL();
-
 // Only do this once, ever.
-$Row = $SQL->Get('Discussion', '', 'asc', 1)->FirstRow(DATASET_TYPE_ARRAY);
-if ($Row) {
-   return;
-}
-
-$DiscussionModel = new DiscussionModel();
+if (!Gdn::SQL()->Get('Discussion', '', 'asc', 1)->FirstRow(DATASET_TYPE_ARRAY)) {
```
Comment (tburry) [inline applications/vanilla/settings/stub.php]: Giant honking if statement is bad. Either keep the early bailout above or refactor to a function.

File: applications/vanilla/settings/stub.php [hunk 2]
```diff
@@ -27,58 +20,81 @@
    <li>Bookmark a discussion (click the star) to get notifications for new comments. You can edit notification settings from your profile.</li>
 </ul>
 Go ahead and edit or delete this discussion, then spread the word to get this place cooking. Cheers!";
-$CommentBody = "This is the first comment on your site and it&rsquo;s an important one.
```
Comment (tburry) [inline applications/vanilla/settings/stub.php]: This is not proper vanilla html formatting and inserts `<br/>` tags into the content. Really, the stub content should be markdown.

File: applications/vanilla/settings/stub.php [hunk 3]
```diff
@@ -6,19 +6,12 @@
...
-
-$DiscussionModel = new DiscussionModel();
+if (!Gdn::SQL()->Get('Discussion', '', 'asc', 1)->FirstRow(DATASET_TYPE_ARRAY)) {
 
-// Prep default content
-$DiscussionTitle = "BAM! You&rsquo;ve got a sweet forum";
-$DiscussionBody = "There&rsquo;s nothing sweeter than a fresh new forum, ready to welcome your community. A Vanilla Forum has all the bits and pieces you need to build an awesome discussion platform customized to your needs. Here&rsquo;s a few tips:
+   // Prep default content
+   $Now = Gdn_Format::ToDateTime();
+   $DiscussionBody = "There&rsquo;s nothing sweeter than a fresh new forum, ready to welcome your community. A Vanilla Forum has all the bits and pieces you need to build an awesome discussion platform customized to your needs. Here&rsquo;s a few tips:
```
Comment (tburry) [inline applications/vanilla/settings/stub.php]: Please no odd "proper" quotes in core.

File: applications/vanilla/settings/stub.php [hunk 4]
```diff
@@ -27,58 +20,81 @@
    <li>Bookmark a discussion (click the star) to get notifications for new comments. You can edit notification settings from your profile.</li>
 </ul>
 Go ahead and edit or delete this discussion, then spread the word to get this place cooking. Cheers!";
-$CommentBody = "This is the first comment on your site and it&rsquo;s an important one.
+   $Discussion = array(
+      'Name' => T('StubDiscussionTitle', "BAM! You&rsquo;ve got a sweet forum"),
```
Comment (tburry) [inline applications/vanilla/settings/stub.php]: There is no way for this translation to take effect since the content is added on install.

**Comment by tburry** [comment]
Really, the stub content should go into an external file such as a php array file or a json file. I'm also concerned about the wording here as I thought things were supposed to be more professional or a "Welcome to your new community" type thing.

**Comment by linc** [comment]
This PR is purely a mechanical change to allow plugging. You're copy editing content that's been this way for over a year now.

**Comment by tburry** [comment]
a) There's other feedback in here. b) I get feedback relating to stuff not specifically in my pull requests all the time.

**Comment by tburry** [comment]
I'm closing this pull request. This is not the way we are going to move forward with stub content.

**Pull request closed without merging**